LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A03_t02.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Assume that DV is an extension type declaration named Name, and
+/// V1 occurs as one of the <type>s in the <interfaces> of DV. In this case we
+/// say that V1 is a superinterface of DV.
+/// ...
+/// Assume that DV has two direct or indirect superinterfaces of the form
+/// W<T1, .. Tk> respectively W<S1, .. Sk>. A compile-time error occurs if Tj is
+/// not equal to Sj for any j in 1 .. k. The notion of equality used here is the
+/// same as with the corresponding rule about superinterfaces of classes.
+///
+/// @description Checks that it is a compile-time error if an extension type has
+/// two superinterfaces of the form `W<T1, .. Tk>`, `W<S1, .. Sk>` and there is
+/// a `j` such that `Tj` is not equal to `Sj`.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+class A<T> {}
+
+extension type IntET(int _) implements int {}
+
+extension type ET<T extends num>(T _) implements num {}
+
+extension type ET1(A<int> _) implements A<num>, A<int> {}
+//                                              ^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+extension type ET2<T extends num>(A<T> _) implements A<num>, A<T> {}
+//                                                             ^
+// [analyzer] unspecified
+// [cfe] unspecified
+extension type ET3<T extends num>(T _) implements ET<T>, ET<num> {}
+
+main() {
+  print(ET1);
+  print(ET2);
+  print(ET3);
```

Comment:
It takes a considerably more complex implementation to detect whether a whole superinterface graph has this kind of conflict, compared to the case where the two occurrences of `A<...>` are in the same syntactic list. So we should have a couple of test cases where they occur in different syntactic lists (like: both occur in superinterfaces; one in the current declaration and another in a superinterface; and a case where the two types aren't syntactically different: `E3 implements E2a<int>, E2b<String>` and `E2a<X> implements A<X>` and `E2b<X> implements A<X>`). Finally, we should also have some cases where the conflict occurs with an extension type (like a copy of the above, but with `extension type A<X> ...`).

I'm not quite sure those things aren't checked already in some other tests, but it would be great if you could add those cases if they are indeed missing at this time.

Reply:
Thanks!